### PR TITLE
[GLUTEN-5341] Fix some Spark 3.5 UTs

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?

Fix some Spark3.5 UTs

## How was this patch tested?

CI

### CODE REVIEW

**Review by zhouyuan — APPROVED** [review]

File: backends-velox/src/test/scala/org/apache/gluten/execution/VeloxHashJoinSuite.scala [hunk 1]
```diff
@@ -88,6 +87,8 @@ class VeloxHashJoinSuite extends VeloxWholeStageTransformerSuite {
       val wholeStages = plan.collect { case wst: WholeStageTransformer => wst }
       if (SparkShimLoader.getSparkVersion.startsWith("3.2.")) {
         assert(wholeStages.length == 1)
+      } else if (SparkShimLoader.getSparkVersion.startsWith("3.5.")) {
```
Comment (ayushi-agarwal) [inline backends-velox/src/test/scala/org/apache/gluten/execution/VeloxHashJoinSuite.scala:90]: Why did it increase to 5 in 3.5? I was also debugging this and saw there were two more exchanges coming in 3.5, shall we debug why there are more exchanges?

Comment (yma11) [reply backends-velox/src/test/scala/org/apache/gluten/execution/VeloxHashJoinSuite.scala:90]: The physical plan in spark3.5 seems changed:
```
*(9) Project [l_partkey#200L]
+- *(9) SortMergeJoin [l_suppkey#201L], [ps_suppkey#156L], Inner
   :- *(6) Sort [l_suppkey#201L ASC NULLS FIRST], false, 0
   :  +- Exchange hashpartitioning(l_suppkey#201L, 5), ENSURE_REQUIREMENTS, [plan_id=300]
   :     +- *(5) Project [l_partkey#200L, l_suppkey#201L]
   :        +- *(5) SortMergeJoin [l_partkey#200L], [p_partkey#123L], Inner
   :           :- *(2) Sort [l_partkey#200L ASC NULLS FIRST], false, 0
   :           :  +- Exchange hashpartitioning(l_partkey#200L, 5), ENSURE_REQUIREMENTS, [plan_id=283]
   :           :     +- *(1) Filter (isnotnull(l_partkey#200L) AND isnotnull(l_suppkey#201L))
   :           :        +- *(1) ColumnarToRow
   :           :           +- BatchScan parquet file:/root/workspace/apache_1/backends-velox/target/scala-2.12/test-classes/tpch-data-parquet-velox/lineitem[l_partkey#200L, l_suppkey#201L] ParquetScan DataFilters: [isnotnull(l_partkey#200L), isnotnull(l_suppkey#201L)], Format: parquet, Location: InMemoryFileIndex(1 paths)[file:/root/workspace/apache_1/backends-velox/target/scala-2.12/test-cl..., PartitionFilters: [], PushedAggregation: [], PushedFilters: [IsNotNull(l_partkey), IsNotNull(l_suppkey)], PushedGroupBy: [], ReadSchema: struct<l_partkey:bigint,l_suppkey:bigint> RuntimeFilters: []
   :           +- *(4) Sort [p_partkey#123L ASC NULLS FIRST], false, 0
   :              +- Exchange hashpartitioning(p_partkey#123L, 5), ENSURE_REQUIREMENTS, [plan_id=292]
   :                 +- *(3) Filter isnotnull(p_partkey#123L)
   :                    +- *(3) ColumnarToRow
   :                       +- BatchScan parquet file:/root/workspace/apache_1/backends-velox/target/scala-2.12/test-classes/tpch-data-parquet-velox/part[p_partkey#123L] ParquetScan DataFilters: [isnotnull(p_partkey#123L)], Format: parquet, Location: InMemoryFileIndex(1 paths)[file:/root/workspace/apache_1/backends-velox/target/scala-2.12/test-cl..., PartitionFilters: [], PushedAggregation: [], PushedFilters: [IsNotNull(p_partkey)], PushedGroupBy: [], ReadSchema: struct<p_partkey:bigint> RuntimeFilters: []
   +- *(8) Sort [ps_suppkey#156L ASC NULLS FIRST], false, 0
      +- Exchange hashpartitioning(ps_suppkey#156L, 5), ENSURE_REQUIREMENTS, [plan_id=309]
         +- *(7) Filter isnotnull(ps_suppkey#156L)
            +- *(7) ColumnarToRow
               +- BatchScan parquet file:/root/workspace/apache_1/backends-velox/target/scala-2.12/test-classes/tpch-data-parquet-velox/partsupp[ps_suppkey#156L] ParquetScan DataFilters: [isnotnull(ps_suppkey#156L)], Format: parquet, Location: InMemoryFileIndex(1 paths)[file:/root/workspace/apache_1/backends-velox/target/scala-2.12/test-cl..., PartitionFilters: [], PushedAggregation: [], PushedFilters: [IsNotNull(ps_suppkey)], PushedGroupBy: [], ReadSchema: struct<ps_suppkey:bigint> RuntimeFilters: []

```
My understanding is that it has 4 exchanges so 5 stages.

Comment (ayushi-agarwal) [reply backends-velox/src/test/scala/org/apache/gluten/execution/VeloxHashJoinSuite.scala:90]: Yes, in 3.4 plan I saw there were only 2 exchanges, there was no exchange after part table and lineitem table scan for their join.

Comment (yma11) [reply backends-velox/src/test/scala/org/apache/gluten/execution/VeloxHashJoinSuite.scala:90]: @ayushi-agarwal Do you want to further dig out the plan change? if so, you can open a ticket dedicated for it. Maybe related with data set or some setting. Let's merge this PR first.

Comment (ayushi-agarwal) [reply backends-velox/src/test/scala/org/apache/gluten/execution/VeloxHashJoinSuite.scala:90]: Sure, I will create a ticket to investigate further. Thanks @yma11

Comment (ayushi-agarwal) [reply backends-velox/src/test/scala/org/apache/gluten/execution/VeloxHashJoinSuite.scala:90]: https://github.com/apache/incubator-gluten/issues/5530

**Pull request merged**